ast/src/analyzed/mod.rs

```diff
@@ -503,6 +503,57 @@ impl<T: FieldElement> Analyzed<T> {
     }
 }
 
+impl<T: Hash> Hash for Analyzed<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        for identity in &self.identities {
+            identity.hash(state);
+        }
+
+        for (key, (_, value)) in self.definitions.iter().sorted_by(|a, b| a.0.cmp(b.0)) {
```

Comment:
Why is the `Symbol` ignored here?

Reply:
Sorry, I left them out when the hash function was specific to the optimization (because they didn't contribute anything to the uniqueness of the hash in that case) and forgot to add them now that it is the general derivation. Fixed.